### PR TITLE
Mod picker grouping fix

### PR DESCRIPTION
**Pull request by nev-r**
causes Charged With Light mods to become 1 category instead of 2, marks artifice armor plug selectable categories as such. all changes previewable in mod picker

### CODE REVIEW

File: src/app/loadout/ModPicker.tsx [hunk 1]
```diff
@@ -154,6 +159,12 @@ function mapStateToProps() {
               maxSelectable,
               plugs,
             };
+            if (
+              maxSelectable === 1 &&
+              armor2PlugCategoryHashes.includes(plugs[0].plug.plugCategoryHash)
+            ) {
```
Comment (ryan-rushton) [inline src/app/loadout/ModPicker.tsx:165]: I wonder if we should just add the `plugSetHash` to `PlugsWithMaxSelectable` and use that? 

I kinda want to turn it into `PlugSetWithMaxSelectable` so we can check the `plugSetHash` as needed. 

Happy to leave it out if you would rather this, but just a thought.

File: src/app/loadout/plug-drawer/PlugDrawer.tsx [hunk 2]
```diff
@@ -12,22 +12,11 @@ import React, { RefObject, useCallback, useEffect, useMemo, useRef, useState } f
...
-  plugs: PluggableInventoryItemDefinition[];
-  maxSelectable: number;
-}
+import PlugSection, { PlugsWithMaxSelectable } from './PlugSection';
 
 interface Props {
-  /**
-   * A list of plug items that come from a PlugSet, along with the maximum number of these plugs that can be chosen.
-   */
   plugsWithMaxSelectableSets: PlugsWithMaxSelectable[];
```
Comment (ryan-rushton) [inline src/app/loadout/plug-drawer/PlugDrawer.tsx:18]: Comment isn't useful?

**Review by ryan-rushton — APPROVED** [review]

File: src/app/loadout/ModPicker.tsx (hunk 1 above)
Comment (nev-r) [reply src/app/loadout/ModPicker.tsx:165]: yeah i am not against waiting to see if something changes, the stakes are pretty low. if you do it with d2ai it's probably wrong at the very beginning of every season

File: src/app/loadout/plug-drawer/PlugDrawer.tsx (hunk 2 above)
Comment (nev-r) [reply src/app/loadout/plug-drawer/PlugDrawer.tsx:18]: felt redundant with the more detailed one attached to the `PlugsWithMaxSelectable` type

**Pull request merged**